File: Week_10/Practicum/Find The Town Judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        in_degree = defaultdict(int)
        out_degree = defaultdict(int)

        for edge in trust:
            out_degree[edge[0]] += 1
            in_degree[edge[1]] += 1

        
        for i in range(1, n + 1):
            if in_degree[i] == n - 1 and out_degree[i] == 0:
                return i

        return -1

File: Week_10/Practicum/Find the Town Judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> trusts(n, 0);
        vector<int> trustHim(n, 0);

        for (auto& t : trust) {
            trusts[t[0] - 1]++;
            trustHim[t[1] - 1]++;
        }

        for (int i = 0; i < n; i++) {
            if (trusts[i] == 0 && trustHim[i] == (n - 1))
                return i + 1;
        }

        return -1;
    }
};